npm_and_yarn/lib/dependabot/npm_and_yarn/file_parser/bun_lock.rb

```diff
@@ -0,0 +1,148 @@
+# typed: strict
+# frozen_string_literal: true
+
+require "yaml"
+require "dependabot/errors"
+require "dependabot/npm_and_yarn/helpers"
+require "sorbet-runtime"
+
+module Dependabot
+  module NpmAndYarn
+    class FileParser < Dependabot::FileParsers::Base
+      class BunLock
+        extend T::Sig
+
+        sig { params(dependency_file: DependencyFile).void }
+        def initialize(dependency_file)
+          @dependency_file = dependency_file
+        end
+
+        sig { returns(T::Hash[String, T.untyped]) }
+        def parsed
+          # Since bun.lock is a JSONC file, which is a subset of YAML, we can use YAML to parse it
+          content = YAML.load(T.must(@dependency_file.content))
+          raise_invalid!("expected to be an object") unless content.is_a?(Hash)
+
+          version = content["lockfileVersion"]
+          raise_invalid!("expected 'lockfileVersion' to be an integer") unless version.is_a?(Integer)
+          raise_invalid!("expected 'lockfileVersion' to be >= 0") unless version >= 0
```

Comment:
perhaps stick to version `0` only, untile newer versions are known to be compatible?